src/ListView.js

```diff
@@ -0,0 +1,265 @@
+import "./todo.css";
+import TopBar from "./TopBar";
+import SubBar from "./SubBar";
+import BottomBar from "./BottomBar";
+import SearchBar from "./SearchBar";
+// import PriorityPopup from "./PriorityPopup";
+import DeleteCompletedPopup from "./DeleteCompletedPopup";
+import ListContents from "./ListContents";
+import Backdrop from "./Backdrop";
+import { useState, useEffect, useRef } from "react";
+import { generateUniqueID } from "web-vitals/dist/modules/lib/generateUniqueID";
+import {
+  useCollectionData,
+  useDocumentData,
+} from "react-firebase-hooks/firestore";
+import {
+  query,
+  setDoc,
+  updateDoc,
+  deleteDoc,
+  doc,
+  orderBy,
+  where,
+  serverTimestamp,
+  collection,
+} from "firebase/firestore";
+import TOP_LEVEL_COLLECTION from "./firestore-config.js";
+
+function ListView(props) {
+  const collectionRef = collection(
+    props.db,
+    TOP_LEVEL_COLLECTION,
+    props.currentList,
+    "items"
+  );
+
+  //   const [sortType, setSortType] = useState("created");
+  const [toScroll, setToScroll] = useState(false);
+
+  const [showCompleted, setShowCompleted] = useState(true);
+
+  const [deleteCompletedPopup, setDeleteCompletedPopup] = useState(false);
+  function handleDeleteCompletedPopup() {
+    setDeleteCompletedPopup(!deleteCompletedPopup);
+  }
+
+  // Use for deleting all completed items
+  const isCheckedQuery = query(collectionRef, where("checked", "==", true));
+  const [checkedData, checkedLoading, checkedError] =
+    useCollectionData(isCheckedQuery);
+
+  const metadataRef = collection(props.db, TOP_LEVEL_COLLECTION);
```

Comment:
Sometimes you appear to use metadata to refer to users and sometimes to the lists. Confusing!